pycs/astro/wl/spherical_mass_mapping.py

```diff
@@ -0,0 +1 @@
        WT_ActiveCoef_e: 2D np.ndarray, active wavelet coefficients E-mode
        WT_ActiveCoef_b: 2D np.ndarray, active wavelet coefficients B-mode
    """
    
    prog="wls_mcalens -m4 "
    # prog_path = ".../cosmostat/build/bin"
    # the cosmostat binary folder has to be in the use system path
    # otherwise you need to hardcode the path in the variable prog:
    #     exemple: prog="Users/starck/git/cosmostat/build/bin/wls_mcalens"
    
    # Create a unique string using the current date and time.
    result=0
    unique_string = datetime.now().strftime('%Y.%m.%d_%H.%M.%S')
    FNin = Prefix + unique_string
    FNout = Prefix + unique_string + '_out'

    file1_in = FNin + '_g1.fits'
    file2_in = FNin + '_g2.fits'
    file3_in = FNin + '_covmat.fits'
    file4_in = FNin + '_cl.fits'

    file1_out = FNout + '_e.fits'
    file2_out = FNout + '_b.fits'
    file3_out = FNout + '_sparse_e.fits'
    file4_out = FNout + '_sparse_b.fits'
    file5_out = FNout + '_active_coef_e.fits'
    file6_out = FNout + '_active_coef_b.fits'

    mrs_write(file1_in, mg1)
    mrs_write(file2_in, mg2)
    mrs_write(file3_in, covmat)
    writefits(file4_in, Cl)

    # print("PROG: ", prog)
    cmd = prog
    if isinstance(opt, type(None)):
        optF=' '
    else:
        optF= opt
    if verbose:
        optF = optF + " -v "
    optF =  optF  + " -s " + str(Nsigma) + " -i " + str(Niter) + " "
    cmd = cmd + " -N " + file3_in  + " -S " + file4_in + " " + optF + " "  + file1_in + " "  +  file2_in + " " + FNout
    if isinstance(Nscale, type(None)):
        ptF =  optF  + " -n " + str(Nscale) + " "
        
    if verbose:
        print ('CMD = ', cmd)

    args = shlex.split(cmd)
    # print('args ', args)
    call(args)

    class res_mcalens():
        in_signal_spectrum=0
        map_e=0
        map_b=0
        map_sparse_e=0
        map_sparse_b=0
        map_wiener_e=0
        map_wiener_b=0        
        map_active_coef_e=0
        map_active_coef_b=0
        WT_ActiveCoef_e=0
        WT_ActiveCoef_b=0
    RES = res_mcalens()
    RES.in_signal_spectrum = Cl
    RES.map_e = mrs_read(file1_out)
    RES.map_b = mrs_read(file2_out)
    RES.map_sparse_e = mrs_read(file3_out)
    RES.map_sparse_b = mrs_read(file4_out)
    RES.map_wiener_e = RES.map_e - RES.map_sparse_e  
    RES.map_wiener_b = RES.map_b - RES.map_sparse_b  
    RES.WT_ActiveCoef_e = readfits(file5_out)
    RES.WT_ActiveCoef_b = readfits(file6_out)   
    RES.map_active_coef_e = RES.WT_ActiveCoef_e.sum(0)
    RES.map_active_coef_b = RES.WT_ActiveCoef_b.sum(0)

    # Return the mr_transform results (and the output file names).
    if remove_files:
        remove(file1_in)
        remove(file2_in)
        remove(file3_in)
        remove(file4_in)
        remove(file1_out)
        remove(file2_out)
        remove(file3_out)
        remove(file4_out)        
        remove(file5_out)
        remove(file6_out)        

    return RES


##########################

```

Comment:
@jstarck There is something wrong with this file. I guess it has to do with the ASCII formatting given the `^M`s that appear. Do you have the original file?

Reply:
This still needs to be fixed if this PR is still valid, otherwise we should close the PR.